spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestUpdate.java

```diff
@@ -1383,36 +1380,35 @@ public void testUpdateWithInvalidAssignmentsAnsi() {
-              SparkException.class,
-              "Null value appeared in non-nullable field",
-              () -> sql("UPDATE %s t SET t.id = NULL", commitTarget()));
-
-          AssertHelpers.assertThrows(
-              "Should complain about writing nulls to a nested column",
-              SparkException.class,
-              "Null value appeared in non-nullable field",
-              () -> sql("UPDATE %s t SET t.s.n1 = NULL", commitTarget()));
-
-          AssertHelpers.assertThrows(
-              "Should complain about writing missing fields in structs",
-              AnalysisException.class,
-              "Cannot find data for the output column `s`.`n2`",
-              () -> sql("UPDATE %s t SET t.s = named_struct('n1', 1)", commitTarget()));
-
-          AssertHelpers.assertThrows(
-              "Should complain about writing invalid data types",
-              AnalysisException.class,
-              "Cannot safely cast",
-              () -> sql("UPDATE %s t SET t.s.n1 = 'str'", commitTarget()));
-
-          AssertHelpers.assertThrows(
-              "Should complain about writing incompatible structs",
-              AnalysisException.class,
-              "Cannot find data for the output column `s`.`n2`.`dn2`",
-              () ->
-                  sql("UPDATE %s t SET t.s.n2 = named_struct('dn3', 1, 'dn1', 2)", commitTarget()));
+          Assertions.assertThatThrownBy(() -> sql("UPDATE %s t SET t.id = NULL", commitTarget()))
+              .as("Should complain about writing nulls to a top-level column")
+              .isInstanceOf(SparkException.class)
+              .hasMessageContaining("Null value appeared in non-nullable field");
+
+          Assertions.assertThatThrownBy(() -> sql("UPDATE %s t SET t.s.n1 = NULL", commitTarget()))
+              .as("Should complain about writing nulls to a nested column")
+              .isInstanceOf(SparkException.class)
+              .hasMessageContaining("Null value appeared in non-nullable field");
+
+          Assertions.assertThatThrownBy(
+                  () -> sql("UPDATE %s t SET t.s = named_struct('n1', 1)", commitTarget()))
+              .as("Should complain about writing missing fields in structs")
+              .isInstanceOf(AnalysisException.class)
+              .hasMessageContaining("Cannot find data for the output column `s`.`n2`");
+
+          Assertions.assertThatThrownBy(() -> sql("UPDATE %s t SET t.s.n1 = 'str'", commitTarget()))
+              .as("Should complain about writing invalid data types")
+              .isInstanceOf(AnalysisException.class)
+              .hasMessageContaining("Cannot safely cast");
+
+          Assertions.assertThatThrownBy(
+                  () ->
+                      sql(
+                          "UPDATE %s t SET t.s.n2 = named_struct('dn3', 1, 'dn1', 2)",
+                          commitTarget()))
+              .as("Should complain about writing incompatible structs")
```

Comment:
I think all of those in this class can be removed

Reply:
I'm thinking these messages would remind on what all cases to verify for this particular class

Reply:
the error msg that is being expected is usally self-explanatory, so we can remove all of those `.as()` definitions